### PR TITLE
Add exception remove person

### PR DESCRIPTION
**Pull request by yongchuann**
Refactor to remove "person" and update it with teacher/student respectively.

Add new exception for both teacher and student.

Update test cases with the new exception

Add new test cases for AddressBookTest.java, ModelManagerTest, LogicManagerTest.java to test for teacher as well

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/student/MedicalHistoryCommand.java [hunk 1]
```diff
@@ -54,9 +54,9 @@ public CommandResult execute(Model model) throws CommandException {
         List<Student> lastShownList = model.getFilteredStudentList();
 
         if (index.getZeroBased() >= lastShownList.size()) {
-            throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            throw new CommandException(Messages.MESSAGE_INVALID_STUDENT_DISPLAYED_INDEX);
         }
-
+        //Todo Remove the typecasting here, use if else instead?
         Person personToEdit = lastShownList.get(index.getZeroBased());
```
Comment (g4ryy) [inline src/main/java/seedu/address/logic/commands/student/MedicalHistoryCommand.java:60]: Perhaps it should be ```Student studentToEdit = lastShownList.get(,,,)```, so no typecasting or if else statements are needed

File: src/main/java/seedu/address/model/person/teacher/exceptions/DuplicateTeacherException.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+import seedu.address.model.person.exceptions.DuplicatePersonException;
+
+/**
+ * Signals that the operation will result in duplicate Teachers
+ * (Teachers are considered duplicates if they have the same identity).
+ */
+public class DuplicateTeacherException extends DuplicatePersonException {
+    public DuplicateTeacherException() {
+        super("Operation would result in duplicate teachers");
+    }
```
Comment (g4ryy) [inline src/main/java/seedu/address/model/person/teacher/exceptions/DuplicateTeacherException.java:12]: Similar to DuplicateStudentException

File: src/main/java/seedu/address/logic/commands/student/MedicalHistoryCommand.java [hunk 3]
```diff
@@ -54,9 +54,9 @@ public CommandResult execute(Model model) throws CommandException {
...
         if (!(personToEdit instanceof Student)) {
             throw new CommandException("You cannot add medical history to a teacher!");
```
[the remark this answers is not in the document]
Comment (random689) [reply src/main/java/seedu/address/logic/commands/student/MedicalHistoryCommand.java]: changed

File: src/main/java/seedu/address/model/person/student/exceptions/DuplicateStudentException.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+import seedu.address.model.person.exceptions.DuplicatePersonException;
+
+/**
+ * Signals that the operation will result in duplicate Students
+ * (Students are considered duplicates if they have the same identity).
+ */
+public class DuplicateStudentException extends DuplicatePersonException {
+    public DuplicateStudentException() {
+        super("Operation would result in duplicate students");
+    }
```
[the remark this answers is not in the document]
Comment (random689) [reply src/main/java/seedu/address/model/person/student/exceptions/DuplicateStudentException.java]: changed

File: src/main/java/seedu/address/model/person/teacher/exceptions/DuplicateTeacherException.java (hunk 2 above)
Comment (random689) [reply src/main/java/seedu/address/model/person/teacher/exceptions/DuplicateTeacherException.java]: changed

Comment (g4ryy) [reply src/main/java/seedu/address/model/person/teacher/exceptions/DuplicateTeacherException.java]: Thank you very much!

**Pull request merged**